spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/procedures/BaseIcebergProcedureCommand.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.catalyst.procedures
+
+import org.apache.iceberg.Table
+import org.apache.iceberg.spark.source.SparkTable
+import org.apache.spark.sql.{AnalysisException, SparkSession}
+import org.apache.spark.sql.connector.catalog.{Identifier, TableCatalog}
+import org.apache.spark.sql.execution.command.RunnableCommand
+
+trait BaseIcebergProcedureCommand extends RunnableCommand {
+
+  protected def loadTable(spark: SparkSession, catalog: TableCatalog, tableName: String): Table = {
```

Comment:
this maybe should be loadIcebergTable? since it fails on non iceberg tables

Reply:
Updated.